tools/test_deleted_renamed_referenced_files

```diff
@@ -5,22 +5,26 @@ success=1
 for FILE in $FILES; do
     if [ -f "$FILE" ] 
     then
-        # if file exists it means it was renamed, and then previous file name is retrieved by git log
+        # if file exists, it means it was renamed. Previous file name is retrieved by git log
         FILE=$(git log --follow -p $FILE | grep 'rename from test' | awk '{print $3}' | head -n 1)
     fi
     if [ -n "$(echo $FILE | grep '\.pm$')" ] 
     then
-        # In case file is a module, module name appears in scheduling files excluding 'tests/' and file extension
+        # If the file is a module, file name appears in scheduling files (excluding part 'tests/' and file extension)
         file_to_verify=$(echo $FILE | sed -E 's/^tests\/(.*)\.pm$/\1/g')
         target_paths='schedule/ products/*/main.pm lib/main_common.pm'
-        # In case file is test_data yaml file, the file given is the same as in schedule files
     elif [ -n "$(echo $FILE | grep '\.ya\?ml$')" ] 
     then
+        # If the file is test_data yaml file, the file name returned from git diff command
+        # is the same as in schedule or test_data files
         file_to_verify="$FILE" 
         target_paths='schedule/ test_data/'
     fi
-    MATCHED_REFERENCE_FILES="$(grep --recursive --ignore-case --files-with-matches "${file_to_verify}\b" $target_paths)"
-    if [[ -n $file_to_verify ]] && [ -n "$(echo $MATCHED_REFERENCE_FILES | grep '\(\.ya\?ml$\|\.pm\)')" ]
+    # Using the -e script option will cause the following command to exit the script with failure in case the file_to_verify
+    # is not referenced anywhere. Using "|| :" to avoid this.
+    MATCHED_REFERENCE_FILES=$(grep --recursive --ignore-case --files-with-matches \
+    "${file_to_verify}\b" $target_paths | grep '\(\.ya\?ml$\|\.pm\)' || :)
```

Comment:
As below we compare against null value (zero length string), I guess we unfortunately need other default ;)

Reply:
What do you mean? Why is it a problem if we compare with null?

Reply:
As far as I've understand, below we check if `MATCHED_REFERENCE_FILES` is null string or not. And those 2 values have different meanings, in case we set it to `:)`, means that it won't be null ever, making second check useless. If we can rely on `file_to_verify`, then we can remove it completely. Do I miss something?

Reply:
The variable gets it's value inside the for loop. So, for every filename that is checked the variable MATCHED_REFERENCE_FILES will get a new value. If the file is found somewhere, the variable will get as value the names of files that mention it. IF not, it will get value NULL, no matter what was the previous value.

Reply:
The confusion is probably that `:` is an executable, equal to `true`, not a string. It doesn't output anything and is equal to `|| true`.

Reply:
Indeed, if you run:
`#!/bin/bash -e

VAR=$(grep -r nonsense /mnt || :)

if [[ -z $VAR ]]
then
echo "VAR is empty"
fi
`
It should print "VAR is empty"

Reply:
@Vogtinator Thanks for clarification, that was indeed my confusion. Something learned today, cheers!